x-pack/plugins/task_manager/server/task_manager.ts

```diff
@@ -456,7 +456,7 @@ export async function awaitTaskRunResult(
                 )
               );
             } else if (isTaskClaimEvent(taskEvent)) {
-              reject(
+              return reject(
```

Comment:
This addresses the Socket timeout - we weren't short circuiting the return here which caused some callbacks to fail weirdly at times.